include/macros.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+#include <string>
+#include <vector>
+
+struct Command {
+	std::vector<std::string> tokens;
+	bool isBackground;
+};
+
+using commands = std::vector<Command>;
```

Comment:
It's better to use the `_t` suffix (like - `commands_t`) when naming types, or name it in capital case. 
Otherwise you might collide with variable names (and `commands` is a very reasonable name for a variable).